### ISSUE TITLE
Replace some C++98isms with C++11isms

### ISSUE THREAD

**Issue by md-5** [znc/znc#816]
Notably we can now use range based for loops, eg:
```c++
            int count = 0;
            for (auto network : m_pUser->GetNetworks()) {
            	for (auto chan : network->GetChans()) {
            		chan->ClearBuffer();
            		count++;
            	}
            }
```
as opposed to the old style:
```c++
const vector<CIRCNetwork*>& vNetworks = m_pUser->GetNetworks();
            vector<CIRCNetwork*>::const_iterator networkIter;
            for (networkIter = vNetworks.begin(); networkIter != vNetworks.end(); ++networkIter) {
                const vector<CChan*>& vChans = (*networkIter)->GetChans();
                vector<CChan*>::const_iterator chanIter;
                for (chanIter = vChans.begin(); chanIter != vChans.end(); ++chanIter) {
                    (*chanIter)->ClearBuffer();
                    count++;
                }
            }
```

**Comment by jpnurmi** [comment]
We've got a lot of C++11 related changes merged. Anything else on the wish list?
- range-based for loops
- override
- nullptr

Loops were changed by hand, override & nullptr were added mostly with help of clang-modernize. Additionally, some auto's could be sprinkled here and there, but I think I'm not going to start systematically doing that. We can do such tweaks when we touch the respective code. Case closed for now?
